garden_ai/client.py

```diff
@@ -122,63 +148,13 @@ def _create_search_authorizer(self):
 
             # now store the tokens and pull out the Groups tokens
             self.auth_key_store.store(response)
-            tokens = response.by_resource_server[SearchClient.resource_server]
-        else:
-            # otherwise, we already did login; load the tokens from that file
-            tokens = self.auth_key_store.get_token_data(SearchClient.resource_server)
-        # construct the RefreshTokenAuthorizer which writes back to storage on refresh
-        authorizer = RefreshTokenAuthorizer(
-            tokens["refresh_token"],
-            self.auth_client,
-            access_token=tokens["access_token"],
-            expires_at=tokens["expires_at_seconds"],
-            on_refresh=self.auth_key_store.on_refresh,
-        )
-        return authorizer
+            tokens = response.by_resource_server[resource_server]
 
-    def _create_garden_authorizer(self):
-        if not self.auth_key_store.file_exists():
-            # do a login flow, getting back initial tokens
-            response = self._do_login_flow()
-
-            if not response:
-                raise AuthException
-
-            # now store the tokens and pull out the Groups tokens
-            self.auth_key_store.store(response)
-
-            tokens = response.by_resource_server[GardenClient.scopes.resource_server]
+            email = extract_email_from_globus_jwt(response.data['id_token'])
+            self._store_user_email(email)
         else:
             # otherwise, we already did login; load the tokens from that file
-            tokens = self.auth_key_store.get_token_data(
-                GardenClient.scopes.resource_server
-            )
-
-        # construct the RefreshTokenAuthorizer which writes back to storage on refresh
-        authorizer = RefreshTokenAuthorizer(
-            tokens["refresh_token"],
-            self.auth_client,
-            access_token=tokens["access_token"],
-            expires_at=tokens["expires_at_seconds"],
-            on_refresh=self.auth_key_store.on_refresh,
-        )
-        return authorizer
-
-    def _authenticate(self):
```

Comment:
DRYed this, _create_garden_authorizer, and create_search_authorizer into `_create_authorizer`